packages/context/src/api/apiServices.ts

```diff
@@ -6,9 +6,12 @@ import { useMemo } from 'react';
 import { AuthenticationContext, useAuthenticationContext } from '../authentication/AuthenticationContextProvider.js';
 import { useApiBaseUrl } from './ApiBaseUrlProvider.js';
 import { Callback } from './callback.js';
+import { randomUUID } from 'crypto';
 
 export type Data = Record<string, unknown> | FormData;
 
+const sessionId = randomUUID();
```

Comment:
This needs to be integrated with `NotificationProvider`.  When a widget subscribes to receive notifications, the `subscribe` needs to know the session id.

Reply:
The current implementation uses the X-Forwarded-Host to restrict who can see messages. That means everyone in the same environment can see it. Now, we uses X-Session-Id to restrict who can see messages.